MANUAL.md

```diff
@@ -0,0 +1,194 @@
+# Testkit manual
+
+Testkit for Exonum blockchain framework, allowing to test service APIs synchronously and in the same process as the testkit.
+
+Below is a simple user guide.
+
+* [Installation](#installation)
+* [Transactions testing](#transactions-testing)
+* [Api testing](#api-testing)
+* [Oracles testing](#oracles-testing)
+* [Configuration changes testing](#configuration-changes-testing)
+
+## Installation
+
+Just add a following line to the `Cargo.toml`:
+
+```toml
+[dev-dependencies]
+exonum-testkit = "0.1.0"
+```
+
+## Simple usage
+
+### Transactions testing
+
+The primary goal of the kind of tests is to check the business logic of your service.
+
+For writing your first test create `tests` directory in according to the cargo
+integration testing [manual][integration-tests].
+After that creates file `transactions.rs` like the following content.
+
+```rust
+extern crate exonum;
+extern crate exonum_testkit;
+extern crate my_service;
+
+use my_service::{MyService, MyTransaction, MySchema};
+use exonum_testkit::TestKitBuilder;
+
+#[test]
+fn test_my_tx() {
+    // Create simple testkit network.
+    let mut testkit = TestKitBuilder::validator()
+        .with_service(MyService::new())
+        .create();
+    // Create transaction.
+    let tx = MyTransaction::new(...);
+    // Commit it into blockchain.
+    testkit.create_block_with_transactions(txvec![tx.clone()]);
+    // Check the expected result
+    let snapshot = testkit.snapshot();
+    let schema = MySchema::new(&snapshot);
+    assert!(schema.is_my_data_checked());
+}
+```
+
+Make sure that you have full coverage of the business logic in the `execute` method of your transactions.
```

Comment:
I'd add that the testkit also allows to check different orderings of transactions, including transactions for multiple services (although the latter is not tested right now, as far as I can see). This could allow to more efficiently test margin cases, which would be difficult to reproduce otherwise.